src/Native/Unix/System.IO.Ports.Native/pal_termios.c

```diff
@@ -0,0 +1,481 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_types.h"
+#include "pal_utilities.h"
+#include <termios.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+#include <errno.h>
+#include <pal_termios.h>
+
+/* This is dup of System/IO/Ports/NativeMethods.cs */
+enum
+{
+    ParityNone = 0,
+    ParityOdd  = 1,
+    ParityEven = 2,
+    ParityMark = 3,
+    ParitySpace = 4
+};
+
+/* System/IO/Ports/Handshake.cs */
+enum
+{
+    HandshakeNone,          // No Flow Control
+    HandshakeSoft,          // Software Flow Control -> XOn/Xoff
+    HandshakeHard,          // Hardware Flow Control -> RTS/CTS
+    HandshakeBoth           // Software & Hardware Flow Control
+};
+
+/* Interop/Unix/Interop.Termios.cs */
+enum
+{
+    SignalDtr = 1,
+    SignalDsr = 2,
+    SignalRts = 3,
+    SignalCts = 4,
+    SignalDcd = 5,
+};
+
+enum
+{
+    AllQueues = 0,
+    ReceiveQueue = 1,
+    SendQueue = 2,
+};
+
+static int32_t SystemIoPortsNative_TermiosGetStatus(intptr_t handle)
+{
+    int fd = ToFileDescriptor(handle);
+    int status = 0;
+    if (ioctl(fd, TIOCMGET, &status) < 0)
+    {
+        return -1;
+    }
+
+    return status;
+}
+
+int32_t SystemIoPortsNative_TermiosGetSignal(intptr_t handle, int32_t signal)
+{
+    int32_t status = SystemIoPortsNative_TermiosGetStatus(handle);
+    if (status == -1)
+    {
+        return -1;
+    }
+
+    switch (signal)
+    {
+    case SignalDtr:
+        return (status & TIOCM_DTR) ? 1 : 0;
+    case SignalDsr:
+        return (status & TIOCM_DSR) ? 1 : 0;
+    case SignalRts:
+        return (status & TIOCM_RTS) ? 1 : 0;
+    case SignalCts:
+        return (status & TIOCM_CTS) ? 1 : 0;
+    case SignalDcd:
+        return (status & TIOCM_CAR) ? 1 : 0;
+    default:
+        return -1;
+   }
+}
+
+int32_t SystemIoPortsNative_TermiosSetSignal(intptr_t handle, int32_t signal, int32_t set)
+{
+    int fd = ToFileDescriptor(handle);
+    int bit;
+
+    switch (signal)
+    {
+    case SignalRts:
+        bit = TIOCM_RTS;
+        break;
+    case SignalDtr:
+        bit = TIOCM_DTR;
+        break;
+    default:
+        errno = EINVAL;
+        return -1;
+    }
+
+    int status = SystemIoPortsNative_TermiosGetStatus(fd);
+    if (status >= 0)
+    {
+        if (set)
+        {
+            status |= bit;
+        }
+        else
+        {
+            status &= ~bit;
+        }
+        return ioctl(fd, TIOCMSET, &status);
+    }
+
+    return -1;
+}
+
+static speed_t SystemIoPortsNative_TermiosSpeed2Rate(int speed)
```

Comment:
Nit: you don't need the prefix for non-exported functions, though it's fine to have it

Reply:
yes, I previously did not have them for internal static functions. 
But I decided it may be better to just keep everything same. 
Thanks for all your feedback.